modules/aruco/test/test_boarddetection.cpp

```diff
@@ -181,8 +185,13 @@ void CV_ArucoBoardPose::run(int) {
     cameraMatrix.at< double >(1, 2) = imgSize.height / 2;
     Mat distCoeffs(5, 1, CV_64FC1, Scalar::all(0));
 
+    double max_dist = 0.4;
+    // aruco3 detection is a bit worse from large distances it seems
+    if (run_with == checkWithParameter::USE_ARUCO3) {
+        max_dist = 0.2;
+    }
```

Comment:
In fact, a too small perimeter marker is created (marker perimeter < 32 * 4). Need set minSideLengthCanonicalImg as 16.